Admin Dashboard

nguyenmanh96 opened the pull request.

Commit:
- Admin dashboard
- Disable the back button

DuongXuanBinh commented on app/Http/Controllers/Admin/AdminDashboardController.php:
```diff
@@ -3,10 +3,55 @@
+    public function adminDashboard()
+    {
+
+        if (Auth::user()->type !== 'Admin') {
```
phần check quyền thế này nên để middleware chứ nhỉ?
Nếu để thế này tất cả các hàm cùng phải dùng à?

DuongXuanBinh commented on app/Http/Controllers/Admin/AdminDashboardController.php:
```diff
@@ -3,10 +3,55 @@
+
+    public function currentWeather()
+    {
+        $apiKey = '7f76453f8ac745bb982164522241205';
```
các phần key này cần ghi vào env hoặc 1 file Enum riêng

nguyenmanh96 replied:
Đã chuyển api key vào .env, và dùng thư viện dotenv để lấy ra api

DuongXuanBinh commented on app/Http/Controllers/Admin/AdminDashboardController.php:
```diff
@@ -3,10 +3,55 @@
         return view('admin.dashboard');
     }
+
+    public function currentTime()
```
current time để làm gì?

nguyenmanh96 replied:
api current time > local time

DuongXuanBinh commented on app/Http/Controllers/Admin/AdminDashboardController.php:
```diff
@@ -3,10 +3,55 @@
+    public function currentWeather()
+    {
+        $apiKey = '7f76453f8ac745bb982164522241205';
+        $city = '21.028511, 105.804817';
```
dùng tên thành phố ko được à? bắt buộc phải dùng lat, long à?

nguyenmanh96 replied:
Đã thay thành 'Hanoi'

DuongXuanBinh commented on app/Http/Controllers/Login/AuthController.php:
```diff
@@ -10,11 +10,15 @@ class AuthController extends Controller
 {
     public function getFormLogin()
     {
+        if (Auth::check()){
```
phần này để làm gì?

nguyenmanh96 replied:
ngăn ngừa khi người dùng khi đã đăng nhập sẽ k thể truy cập vào trang login bằng domain

DuongXuanBinh commented on app/Http/Controllers/Student/DashboardController.php:
```diff
@@ -3,11 +3,15 @@
 class DashboardController extends Controller
 {
-    public function Dashboard(){
+    public function Dashboard()
```
tương tự màn admin, có cần check type ở đây ko? 
tên hàm sai convention

nguyenmanh96 replied:
đã cho vào middleware và sửa tên hàm.

DuongXuanBinh commented on composer.json:
```diff
@@ -6,7 +6,7 @@
     "require": {
         "php": "^8.1",
-        "guzzlehttp/guzzle": "^7.2",
+        "guzzlehttp/guzzle": "^7.8",
```
sao lại update cái này?

nguyenmanh96 replied:
e clone bằng driver nên nhặt những file thay đổi > sang máy mới update lại composer thì phải

DuongXuanBinh commented on public/js/admin/layout.js:
```diff
@@ -1 +1,92 @@
+}
+
+animateStars();
+
```
code này lấy về có hiểu ko? Tự viết lại nhé
Ngoài ra hiện tại đang chạy mỗi 1s, đề bài là chạy mỗi 5s

DuongXuanBinh commented on resources/views/admin/dashboard.blade.php:
```diff
@@ -2,28 +2,28 @@
+            <div class="blush blush2"></div>
+        </div>
+
+        <div class="orbit">
```
tên class lạ thế? copy đâu à

nguyenmanh96 replied:
tên class thư viện > copy

DuongXuanBinh commented on resources/views/layouts/admin/header.blade.php:
```diff
@@ -1,37 +1,78 @@
+                </a>
+                <hr class="atom-toolbar-spacer">
+                <a href="#" class="btn btn-default">
+                    <span class="octicon octicon-file-directory"></span>
```
tên class lạ nha

nguyenmanh96 replied:
tên class thư viện

DuongXuanBinh commented on resources/views/layouts/admin/layout.blade.php:
```diff
@@ -7,15 +7,18 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Document</title>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.2/css/all.min.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/Primer/3.0.1/css/primer.css">
```
tên class trên là của thư viện này à?

nguyenmanh96 replied:
Vâng

DuongXuanBinh commented on public/js/admin/layout.js:
```diff
@@ -1 +1,92 @@
+$(document).ready(function () {
+    setInterval(function () {
+        $.ajax({
+            url: 'http://qlhs.com/admin/get-time',
```
không nên dùng tên domain như thế này
Trường hợp đây là local, sau này lên staging domain khác lại phải đi tìm từng file để sửa

nguyenmanh96 replied:
đã đổi > lấy time từ máy khách

DuongXuanBinh commented on resources/views/admin/dashboard.blade.php:
```diff
@@ -2,28 +2,28 @@
+    <div class="sky">
+        <div class="moon-hi" >
+            <span>Hello </span>
+            <span>{{\Illuminate\Support\Facades\Auth::user()->type}}</span>
```
Nên là chào user name chứ ko chào type

nguyenmanh96 replied:
đã sửa ->name

nguyenmanh96 committed:
Fix bug admin dashboard
- Add middleware
- Api key hidden
- Rename function, variable, value
- Change the method to get the time
